src/Core/Style.js

```diff
@@ -131,23 +138,79 @@ class Style {
+            this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-z-order'] || 'auto', zoom);
+            if (this.text.zOrder == 'auto') {
+                this.text.zOrder = readVectorProperty(layer.layout && layer.layout['symbol-sort-key'] || 'Y', zoom);
+            } else if (this.text.zOrder == 'viewport-y') {
+                this.text.zOrder = 'Y';
+            } else if (this.text.zOrder == 'source') {
+                this.text.zOrder = 0;
+            }
+
+            // position
+            this.text.anchor = readVectorProperty(layer.layout && layer.layout['text-anchor'] || 'center', zoom);
+            this.text.offset = readVectorProperty(layer.layout && layer.layout['text-offset'] || [0, 0], zoom);
+            this.text.padding = readVectorProperty(layer.layout && layer.layout['text-padding'] || 2, zoom);
+            this.text.size = readVectorProperty(layer.layout && layer.layout['text-size'] || 16, zoom);
+            this.text.placement = readVectorProperty(layer.layout && layer.layout['symbol-placement'] || 'point', zoom);
+            this.text.rotation = readVectorProperty(layer.layout && layer.layout['text-rotation-alignment'] || 'auto', zoom);
+
+            // content
+            this.text.field = readVectorProperty(layer.layout && layer.layout['text-field'] || '', zoom);
+            this.text.wrap = readVectorProperty(layer.layout && layer.layout['text-max-width'] || 10, zoom);
+            this.text.spacing = readVectorProperty(layer.layout && layer.layout['text-letter-spacing'] || 0, zoom);
+            this.text.transform = readVectorProperty(layer.layout && layer.layout['text-transform'] || 'none', zoom);
+            this.text.justify = readVectorProperty(layer.layout && layer.layout['text-justify'] || 'center', zoom);
+
+            // appearance
+            const { color, opacity } = rgba2rgb(readVectorProperty(layer.paint && layer.paint['text-color'] || '#000000', zoom));
+            this.text.color = color;
+            this.text.opacity = readVectorProperty(layer.paint && layer.paint['text-opacity'], zoom) || (opacity !== undefined && opacity) || 1.0;
+            this.text.font = readVectorProperty(layer.layout && layer.layout['text-font'] || ['Open Sans Regular', 'Arial Unicode MS Regular'], zoom);
+            this.text.halo = {};
+            this.text.halo.color = readVectorProperty(layer.paint && layer.paint['text-halo-color'] || '#000000', zoom);
+            this.text.halo.width = readVectorProperty(layer.paint && layer.paint['text-halo-width'] || 0, zoom);
+            this.text.halo.blur = readVectorProperty(layer.paint && layer.paint['text-halo-blur'] || 0, zoom);
+
+            // additional icon
+            const iconSrc = readVectorProperty(layer.layout && layer.layout['icon-image'], zoom);
+            if (iconSrc) {
+                let size = readVectorProperty(layer.layout && layer.layout['icon-size'], zoom);
+                if (size == undefined) { size = 1; }
```

Comment:
`const size = readVectorProperty(layer.layout && layer.layout['icon-size'] || 1, zoom);`

Reply:
I can't, because it can be equal to `0`.